saq/queue.py

```diff
@@ -46,6 +60,13 @@ def __init__(self, redis, name="default", dump=None, load=None):
         self._sweep = self.namespace("sweep")
         self._dump = dump or json.dumps
         self._load = load or json.loads
+        self._before_enqueue_callbacks = {}
```

Comment:
to be consistent with worker, should we call it _before_enqueue, worker has _before_process

Reply:
should we just use a set? then you can just add and remove from the set?

Reply:
A set requires that the object is hashable. That seems like an unnecessary constraint.

As for the name, I'm cool with just `_before_enqueue`. Note, this one supports _multiple_ callbacks. That's because I'm planning to also use this functionality to support distributed tracing.

Reply:
ok